packages/assets-controllers/src/TokensController.ts

```diff
@@ -700,49 +704,84 @@ export class TokensController extends BaseController<
    * Parameters will be validated according to the asset type being watched.
    *
    * @param options - The method options.
-   * @param options.asset - The asset to be watched. For now only ERC20 tokens are accepted.
    * @param options.type - The asset type.
+   * @param options.address - The contract address of the asset.
+   * @param options.image - An optional URL to the asset's icon.
    * @param options.interactingAddress - The address of the account that is requesting to watch the asset.
    * @param options.networkClientId - Network Client ID.
-   * @returns Object containing a Promise resolving to the suggestedAsset address if accepted.
+   * @returns A promise that resolves if the asset was watched successfully, and rejects otherwise.
    */
   async watchAsset({
-    asset,
     type,
+    address,
+    image,
     interactingAddress,
     networkClientId,
   }: {
-    asset: Token;
     type: string;
+    address: string;
+    image: string;
     interactingAddress?: string;
     networkClientId?: NetworkClientId;
   }): Promise<void> {
     if (type !== ERC20) {
       throw new Error(`Asset of type ${type} not supported`);
     }
 
+    if (!address) {
+      throw rpcErrors.invalidParams('Address must be specified');
+    }
+
+    if (!isValidHexAddress(address)) {
+      throw rpcErrors.invalidParams(`Invalid address "${address}".`);
+    }
+
+    const details = await this.getTokenStandardAndDetails(
+      address,
+      undefined,
+      undefined,
+      networkClientId,
+    );
+
+    if (details.standard !== ERC20) {
+      throw rpcErrors.invalidParams(
+        `Contract ${address} must match type ${type}, but was detected as ${details.standard}`,
+      );
+    }
+
+    const symbol = details.symbol ?? '';
```

Comment:
It should not be possible for this to be undefined because we've guarded `standard === ERC20`.  However, `symbol` and `decimals` are optional in the [ERC-20 spec](https://eips.ethereum.org/EIPS/eip-20).  So it's not clear it's correct that our standard detection, types, and apps expect them.

Reply:
I will try to improve the case where `symbol` and `decimals` aren't defined in the contract.  In this case, we can keep accepting values provided by the dapp.